### PR TITLE
Refactor data generation for efficiency and better DX

### PR DESCRIPTION
**Pull request by atvaccaro**
# Description

Pre-requisite for https://github.com/cal-itp/reports/issues/234

* Refactor so we are no longer dependent on `make` (just use it for saving commands, not for actual execution/parallelization)
* Caches un-filtered BigQuery queries to drastically reduce runtime
* Fixes docker/docker-compose, uses poetry for dependencies, and sets up devcontainer (https://github.com/cal-itp/data-infra/issues/2420)
* Updates documentation as relevant

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation

## How has this been tested?
Ran through this with Charlie today. We're going to wait until after this month's reports, but merge ASAP after so @owades is unblocked.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ This repository is set up in two pieces:
 
 ## To Get Started
 
+<<<<<<< HEAD
```
Comment (owades) [inline README.md:14]: remove

**Review by owades — COMMENTED** [review]
I can't offer much feedback without first testing - let me know if you want me to do so now. I did ID one line of text to be removed, though.

**Comment by atvaccaro** [comment]
> I can't offer much feedback without first testing - let me know if you want me to do so now. I did ID one line of text to be removed, though.

Thanks! And I think we can test tomorrow when @charlie-costanzo walks us through the reports process.

**Review by charlie-costanzo — APPROVED** [review]
LGTM! Thank you for taking care of this!

**Pull request merged**